src/auth/awss3/core/qgsauthawss3method.cpp

```diff
@@ -134,6 +137,85 @@ bool QgsAuthAwsS3Method::updateNetworkRequest( QNetworkRequest &request, const Q
   return true;
 }
 
+bool QgsAuthAwsS3Method::updateDataSourceUriItems( QStringList &connectionItems, const QString &authcfg,
+    const QString &dataprovider )
+{
+  Q_UNUSED( dataprovider )
+  const QMutexLocker locker( &mMutex );
```

Comment:
I don't think we need to lock mutex has it protect static sAuthConfigCache variable and its not used in this method (It's used in getMethodConfig but already locked)

Reply:
I was following this as an example:
https://github.com/qgis/QGIS/blob/master/src/auth/basic/core/qgsauthbasicmethod.cpp#L96

but if its not required I can remove it

Reply:
Indeed, but I see no access to member or static variables in that method so I think you can safely remove it